app/services/agency_identity_linker.rb

```diff
@@ -0,0 +1,49 @@
+class AgencyIdentityLinker
+  def initialize(sp_identity)
+    @sp_identity = sp_identity
+    @agency_id = nil
+  end
+
+  def link_identity
```

Comment:
My understanding of the flow looking at this code:

- Check for an agency identity with the same uuid as the identity, return if that exists
- Get agency id from the service provider, return uuid of agency identity for the SP and user if it exists
- Create an agency identity for the user and agency and return if the SP has an agency id
- If all else fails, create an agency identity with the same UUID as the base identity and return that.

I suppose you create the agency identity without `agency_id` so you can depend on the result of `link_identity` being an `AgencyIdentity` object.

Reply:
correct